coalahtml/tree.py

```diff
@@ -0,0 +1,84 @@
+from collections import OrderedDict
+from queue import Queue
+import os
+
+
+class Node:
+    """
+    Node class
+    """
+    count = 1
+
+    def __init__(self, path):
+        self.child = {}
+        self.parent = None
+        self.path = path
+        self.abspath = ""
+        self._id = Node.count
+        Node.count += 1
+
+
+class Tree:
+
+    def __init__(self, file_list):
+        """
+        Create a root and build the graph.
+        """
+        self.root = Node(os.path.commonprefix(file_list))
+        self.root.abspath = self.root.path
+        self.fl_split = Tree._split_paths(file_list)
+        self._build(self.fl_split)
+
+    @staticmethod
+    def _split_paths(path_list):
+        return [path.split('/') for path in path_list]
```

Comment:
Won't having just os.sep would work?
ᐧ

On Wed, May 25, 2016 at 11:42 PM, Attila <notifications@github.com> wrote:

> In coalahtml/tree.py
> <https://github.com/coala-analyzer/coala-html/pull/46#discussion_r64623495>
> :
>
> > +
> > +
> > +class Tree:
> > +
> > +    def __init__(self, file_list):
> > +        """
> > +        Create a root and build the graph.
> > +        """
> > +        self.root = Node(os.path.commonprefix(file_list))
> > +        self.root.abspath = self.root.path
> > +        self.fl_split = Tree._split_paths(file_list)
> > +        self._build(self.fl_split)
> > +
> > +    @staticmethod
> > +    def _split_paths(path_list):
> > +        return [path.split('/') for path in path_list]
>
> Probably you can use os.path.split with some tweaking. Take a look at
> http://stackoverflow.com/questions/3167154/how-to-split-a-dos-path-into-its-components-in-python/3167684#3167684
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/coala-analyzer/coala-html/pull/46/files/3ccae34bd5a37e74f317493ac89a4dd2ce56133d..661e49c95350dc09646e3a4b15df6cf3fe2430de#r64623495>
>